### ISSUE TITLE
Allow different spins to have different lengths

### ISSUE THREAD

**Issue by fretchen** [Qiskit-Extensions/qiskit-cold-atom#9]
Suppose the following circuit:

```
qc = QuantumCircuit(2)
qc.lx(np.pi, [0])

qc.append(LxLyGate(gamma=np.pi), [0, 1])
qc.measure_all()
counts = backend.run(qc, spin=1/2).result().get_counts()
```

To implement a number of mixture experiments it would be necessary to set the length of each individual spin, which would modify the last line too:

```
qc = QuantumCircuit(2)
qc.lx(np.pi, [0])

qc.append(LxLyGate(gamma=np.pi), [0, 1])
qc.measure_all()
counts = backend.run(qc, spin=[5, 1/2]).result().get_counts()
```

No idea which part of the code has to be changed for that and if it is even possible right now.

**Comment by eggerdj** [comment]
FYI https://github.com/Qiskit/qiskit-nature/issues/383

**Comment by fretchen** [comment]
Agreed. As long as the issue remains in qiskit-nature it is likely a good approach to 'simply' implement this an indivdual provider back-end as it starts to be on the more specialized use-case.

**Comment by LaurinFischer** [comment]
Yes, Daniel is right that this limitation from the `SpinOp` is what prompted us to set this constant for now. We could build the necessary matrices within the `SpinCircuitSolver` ourselves if this turns out to be a difficult extension for the base `SpinOp` from qiskit nature.
